tern/__main__.py

```diff
@@ -35,6 +35,13 @@
 
 logger.addHandler(log_handler)
 
+try:
+    from tern.report import report
+except Exception as e:
+    logger.addHandler(console)
+    logger.critical("Couldn't set up the report module:\n\t%s", e)
+    sys.exit(1)
+
```

Comment:
On further review, I don't think this is a good way to solve this issue long term. Docker isn't supposed to be a hard dependency. It's only needed when you are specifically using the Docker API to manage images. I know the code looks like it's servicing only containers created by Docker but long-term, it needs to account for other image types. So I think rather than do the check here, we should do the check during execution.

Reply:
This requires a lot more messing around with `tern/utils/container.py` because the client is global. Rather than have that piece of code global, we can turn that into a function that returns the client object. The have the other functions accept an extra client object if there is any Docker interaction. That way, `execute_dockerfile` and `execute_docker_image` can do a check for the client and exit gracefully on error.

Reply:
Completely agree. I went down this path as I didn't want to refactor a lot of things without knowing what's the idea long term. I was thinking that, for now, this would achieve what we want on a less than 30-line diff.

We can close it and I can take a stab at refactoring this docker module to handle things more appropriately, or we could merge it and have a ticket about this that I can also gladly work on :+1: 
Thanks!

Reply:
I'd greatly appreciate it if you can work on a long term solution! I will close this PR.